pkgs/top-level/all-packages.nix

```diff
@@ -21375,7 +21375,11 @@ with pkgs;
 
   hasura-cli = callPackage ../servers/hasura/cli.nix { };
 
-  hbase = callPackage ../servers/hbase {};
+  inherit (callPackage ../servers/hbase {}) hbase_1_7_1 hbase_2_4_11 hbase_3_0_0_a2;
+  hbase1 = hbase_1_7_1;
```

Comment:
Please name them hbase_1, hbase_2 and hbaee_3 otherwise we need to update more things than we need with every update. Also hbase[1-3] can be dropped.

Reply:
This is to prevent silent minor version upgrades. If you're spinning up ad-hoc clusters and don't care about the precise minor version being used, hbase{1..3} can be used. If you're running a non-ephemeral cluster for long term use, you want to be sure that bumping nixpkgs doesn't silently change the version of HBase/Hadoop. This convention is also followed by other hadoop-related packages in nixpkgs.

Reply:
> This is to prevent silent minor version upgrades.

Then why the full version with a another number after patch is exposed? If you want to be absolutely sure that nothing break pin your nixpkgs or vendor that one package. We do not expose package with full version names in the attribute or all versions of a package.

> This is also not that different from what python does with python->python2/3->python310/311/27

It is different. We have python311, not python3.11.0a7.
PS: python2 got removed and if it wasn't for nixops more of python27 would also be hone. Also python is the internal alias in pythonPackages to the current interpreter.

Reply:
> Then why the full version with a another number after patch is exposed?

We could drop the maintenance version number from the name and just have major/minor

> PS: python2 got removed

doesn't look like it
```
[illustris@illustris-thinkpad:~]$ nix run github:nixos/nixpkgs/master#python2 -- --version
Python 2.7.18
[illustris@illustris-thinkpad:~]$ nix run github:nixos/nixpkgs/master#python27 -- --version
Python 2.7.18
```

> alias

I guess the major versions could be made into aliases, but that would go against your original point about having to update more things. Version changes would need both aliases and all_packages to be updated.